network/src/peer_manager/mod.rs

```diff
@@ -171,12 +171,10 @@ impl PeerManagerRequestSender {
             res_tx,
             timeout,
         };
-        self.inner
-            .push(
-                (peer_id, protocol),
-                PeerManagerRequest::SendRpc(peer_id, request),
-            )
-            .unwrap();
+        self.inner.push(
```

Comment:
agree, without a comment, it is unclear if this unwrap is intentional, for example, the node could be so overloaded that we want to explicitly crash the service. so this seems okay.